utils/tests_fetcher.py

```diff
@@ -0,0 +1,427 @@
+# coding=utf-8
+# Copyright 2021 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import collections
+import os
+import re
+from contextlib import contextmanager
+from pathlib import Path
+
+from git import Repo
+
+
+# This script is intended to be run from the root of the repo but you can adapt this constant if you need to.
+PATH_TO_TRANFORMERS = "."
+
+
+@contextmanager
+def checkout_commit(repo, commit_id):
+    """
+    Context manager that checks out a commit in the repo.
+    """
+    current_head = repo.head.commit if repo.head.is_detached else repo.head.ref
+
+    try:
+        repo.git.checkout(commit_id)
+        yield
+
+    finally:
+        repo.git.checkout(current_head)
+
+
+def clean_code(content):
+    """
+    Remove docstrings, empty line or comments from `content`.
+    """
+    # fmt: off
+    # Remove docstrings by splitting on triple " then triple ':
+    splits = content.split('\"\"\"')
```

Comment:
do we also need to split on `r\"\"\"`? (such as docstring that have `r""""`) or is that fine

Reply:
The code is not analyzed after, so the only downside is that if someone removes the `r`, the diff will count as real code modification and trigger some tests. I don't think we need a special case for that.